src/test/java/com/hubspot/jinjava/tree/FailOnUnknownTokensTest.java

```diff
@@ -40,4 +41,21 @@ public void itReplaceTokensWithoutException() {
     String renderedTemplate = jinjava.render(template, context);
     assertThat(renderedTemplate).isEqualTo("hello test and test1");
   }
+
+  @Test
+  public void itReplacesTokensInContextButThrowsExceptionForOthers() {
+    Map<String, String> context = new HashMap<>();
+    context.put("animal", "lamb");
+    context.put("fruit", "apple");
+    String template = "{{ name }} has a {{ animal }}";
+    try{
+      jinjava.render(template, context);
+    } catch (Exception ex) {
+      assertThat(ex.getMessage()).containsIgnoringCase("unknown token found");
```

Comment:
this doesn't guarantee that the exception is thrown. You can use `assertThatThrownBy` for that.

Reply:
okay, will make the changes